src/librustdoc/html/render/mod.rs

```diff
@@ -2218,8 +2218,17 @@ fn extra_info_tags(item: &clean::Item, parent: &clean::Item) -> String {
 
     // The trailing space after each tag is to space it properly against the rest of the docs.
     if let Some(depr) = &item.deprecation {
+        // FIXME: this is awful, isn't there a better way?
+        let symbol;
+        let since = match depr.since {
+            Some(s) => {
+                symbol = s.as_str();
+                Some(&*symbol)
+            }
+            None => None,
+        };
```

Comment:
I assume you have to do this for lifetime reasons? I assume this fails to compile:

```suggestion
        let since = match depr.since {
            Some(s) => {
                Some(s.as_str())
            }
            None => None,
        };
```

Can you give the error that happens with the code I suggested?

Reply:
Oh hmm this does work with `as_deref()`. I tried `&since` and `since.map(|s| &*s)` before, I guess as_deref does lifetime magic?

Reply:
The source for `Option::as_deref()` (I think that's the method being called?) is:

```rust
    pub fn as_deref(&self) -> Option<&T::Target> {
        self.as_ref().map(|t| t.deref())
    }
```

So it doesn't seem to be magic.

Reply:
Ahhh, the `as_ref()` is what I was missing in my code. Thanks!